hie-plugin-api/Haskell/Ide/Engine/PluginsIdeMonads.hs

```diff
@@ -414,6 +405,46 @@ getVirtualFile uri = do
     Just lf -> liftIO $ Core.getVirtualFileFunc lf (toNormalizedUri uri)
     Nothing -> return Nothing
 
+-- | Persist a virtual file as a temporary file in the filesystem.
+-- If the virtual file associated to the given URI does not exist then
+-- the FilePath parsed from the URI is returned.
+persistVirtualFile :: (MonadIde m, MonadIO m) => Uri -> m FilePath
+persistVirtualFile uri = fromMaybe (error "persist")  <$> getPersistedFile uri
```

Comment:
Either this function should be deleted or the call to `error` replaced with a proper exception which can be caught/handled. I don't know where it is used anymore.

Reply:
I imagine it's just unused.

Reply:
It is unused and removed